chainer/functions/reshape.py

```diff
@@ -1,13 +1,31 @@
+import numpy
+
 from chainer import function
+from chainer.utils import type_check
 
 
 class Reshape(function.Function):
+    type_check_prod = type_check.Variable(numpy.prod, 'prod')
```

Comment:
Do not write anything above the docstring. How about moving it to module level and name it like `_type_check_prod`?